include/reactor-uc/scheduler.h

```diff
@@ -8,19 +8,9 @@ typedef struct Scheduler Scheduler;
 typedef struct Environment Environment;
 
 struct Scheduler {
-  Environment *env;
-  EventQueue event_queue;
-  ReactionQueue reaction_queue;
-  // The following two fields are used to implement a linked list of Triggers
-  // that are registered for cleanup at the end of the current tag.
-  Trigger *cleanup_ll_head;
-  Trigger *cleanup_ll_tail;
-  bool leader;          // Whether this scheduler is the leader in a federated program and selects the start tag.
-  instant_t start_time; // The physical time at which the program started.
-  interval_t duration;  // The duration after which the program should stop.
-  tag_t stop_tag;       // The tag at which the program should stop. This is set by the user or by the scheduler.
-  tag_t current_tag;    // The current logical tag. Set by the scheduler and read by user in the reaction bodies.
-  bool keep_alive;      // Whether the program should keep running even if there are no more events to process.
+  interval_t start_time;
+  interval_t duration; // The duration after which the program should stop.
+  bool keep_alive;     // Whether the program should keep running even if there are no more events to process.
 
   /**
    * @brief Schedules an event on trigger at a specified tag. This function will
```

Comment:
prepare_timestep, cleanup_timestep, run_timestep seems to only be needed by the dynamic scheduler? Should probably be moved from the generic API?